<Type Name="ForeignKeyReferenceAlreadyHasValueException" FullName="System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException">
  <Metadata><Meta Name="ms.openlocfilehash" Value="0ec95a295b6394c5cc4b73d4441578fda4d5b851" /><Meta Name="ms.sourcegitcommit" Value="11d168140aa8fade0768c2a9dde3e3bcacfdfb7d" /><Meta Name="ms.translationtype" Value="MT" /><Meta Name="ms.contentlocale" Value="zh-CN" /><Meta Name="ms.lasthandoff" Value="05/27/2020" /><Meta Name="ms.locfileid" Value="83965432" /></Metadata><TypeSignature Language="C#" Value="public class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="ILAsm" Value=".class public auto ansi beforefieldinit ForeignKeyReferenceAlreadyHasValueException extends System.InvalidOperationException" />
  <TypeSignature Language="DocId" Value="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" />
  <TypeSignature Language="VB.NET" Value="Public Class ForeignKeyReferenceAlreadyHasValueException&#xA;Inherits InvalidOperationException" />
  <TypeSignature Language="C++ CLI" Value="public ref class ForeignKeyReferenceAlreadyHasValueException : InvalidOperationException" />
  <TypeSignature Language="F#" Value="type ForeignKeyReferenceAlreadyHasValueException = class&#xA;    inherit InvalidOperationException" />
  <AssemblyInfo>
    <AssemblyName>System.Data.Linq</AssemblyName>
    <AssemblyVersion>3.5.0.0</AssemblyVersion>
    <AssemblyVersion>4.0.0.0</AssemblyVersion>
  </AssemblyInfo>
  <Base>
    <BaseTypeName>System.InvalidOperationException</BaseTypeName>
  </Base>
  <Interfaces />
  <Docs>
    <summary><span data-ttu-id="9417a-101">表示在已经加载实体的情况下尝试更改外键时发生的错误。</span><span class="sxs-lookup"><span data-stu-id="9417a-101">Represents errors that occur when an attempt is made to change a foreign key when the entity is already loaded.</span></span></summary>
    <remarks>To be added.</remarks>
  </Docs>
  <Members>
    <MemberGroup MemberName=".ctor">
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Docs>
        <summary><span data-ttu-id="9417a-102">初始化 <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="9417a-102">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class.</span></span></summary>
      </Docs>
    </MemberGroup>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException ();" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor() cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor" />
      <MemberSignature Language="VB.NET" Value="Public Sub New ()" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException();" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters />
      <Docs>
        <summary><span data-ttu-id="9417a-103">使用说明错误的系统提供的消息初始化 <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="9417a-103">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a system-supplied message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="9417a-104">此构造函数将新实例的 <xref:System.Exception.Message%2A> 属性初始化为系统提供的消息，该消息描述诸如“DefaultMessageDisplayedByParameterlessCtorWriterMustSupply”的错误。</span><span class="sxs-lookup"><span data-stu-id="9417a-104">This constructor initializes the <xref:System.Exception.Message%2A> property of the new instance to a system-supplied message that describes the error, such as "DefaultMessageDisplayedByParameterlessCtorWriterMustSupply".</span></span> <span data-ttu-id="9417a-105">此消息会考虑当前系统区域性。</span><span class="sxs-lookup"><span data-stu-id="9417a-105">This message takes into account the current system culture.</span></span>  
  
 <span data-ttu-id="9417a-106">下表显示了 <xref:System.InsufficientMemoryException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="9417a-106">The following table shows the initial property values for an instance of <xref:System.InsufficientMemoryException>.</span></span>  
  
|<span data-ttu-id="9417a-107">属性</span><span class="sxs-lookup"><span data-stu-id="9417a-107">Property</span></span>|<span data-ttu-id="9417a-108">值</span><span class="sxs-lookup"><span data-stu-id="9417a-108">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="9417a-109">`null` 列中的一个值匹配。</span><span class="sxs-lookup"><span data-stu-id="9417a-109">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="9417a-110">本地化的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="9417a-110">The localized error message string.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="9417a-111">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="9417a-111">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException message" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="9417a-112">描述异常的消息。</span><span class="sxs-lookup"><span data-stu-id="9417a-112">The message that describes the exception.</span></span> <span data-ttu-id="9417a-113">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span><span class="sxs-lookup"><span data-stu-id="9417a-113">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <summary><span data-ttu-id="9417a-114">使用说明错误的指定消息初始化 <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="9417a-114">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified message that describes the error.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="9417a-115">下表显示了 <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="9417a-115">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="9417a-116">属性</span><span class="sxs-lookup"><span data-stu-id="9417a-116">Property</span></span>|<span data-ttu-id="9417a-117">值</span><span class="sxs-lookup"><span data-stu-id="9417a-117">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="9417a-118">`null` 列中的一个值匹配。</span><span class="sxs-lookup"><span data-stu-id="9417a-118">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="9417a-119">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="9417a-119">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="9417a-120">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="9417a-120">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
    <Member MemberName=".ctor">
      <MemberSignature Language="C#" Value="public ForeignKeyReferenceAlreadyHasValueException (string message, Exception innerException);" />
      <MemberSignature Language="ILAsm" Value=".method public hidebysig specialname rtspecialname instance void .ctor(string message, class System.Exception innerException) cil managed" />
      <MemberSignature Language="DocId" Value="M:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException.#ctor(System.String,System.Exception)" />
      <MemberSignature Language="VB.NET" Value="Public Sub New (message As String, innerException As Exception)" />
      <MemberSignature Language="C++ CLI" Value="public:&#xA; ForeignKeyReferenceAlreadyHasValueException(System::String ^ message, Exception ^ innerException);" />
      <MemberSignature Language="F#" Value="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException : string * Exception -&gt; System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" Usage="new System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException (message, innerException)" />
      <MemberType>Constructor</MemberType>
      <AssemblyInfo>
        <AssemblyName>System.Data.Linq</AssemblyName>
        <AssemblyVersion>3.5.0.0</AssemblyVersion>
        <AssemblyVersion>4.0.0.0</AssemblyVersion>
      </AssemblyInfo>
      <Attributes>
        <Attribute FrameworkAlternate="netframework-4.0">
          <AttributeName>System.Runtime.TargetedPatchingOptOut("Performance critical to inline this type of method across NGen image boundaries")</AttributeName>
        </Attribute>
      </Attributes>
      <Parameters>
        <Parameter Name="message" Type="System.String" />
        <Parameter Name="innerException" Type="System.Exception" />
      </Parameters>
      <Docs>
        <param name="message"><span data-ttu-id="9417a-121">描述异常的消息。</span><span class="sxs-lookup"><span data-stu-id="9417a-121">The message that describes the exception.</span></span> <span data-ttu-id="9417a-122">此构造函数的调用方需要确保此字符串已针对当前系统区域性进行了本地化。</span><span class="sxs-lookup"><span data-stu-id="9417a-122">The caller of this constructor is required to ensure that this string has been localized for the current system culture.</span></span></param>
        <param name="innerException"><span data-ttu-id="9417a-123">引起当前异常的异常。</span><span class="sxs-lookup"><span data-stu-id="9417a-123">The exception that is the cause of the current exception.</span></span> <span data-ttu-id="9417a-124">如果 <paramref name="innerException" /> 参数不为 <see langword="null" />，则当前异常将在处理内部异常的 <see langword="catch" /> 块中引发。</span><span class="sxs-lookup"><span data-stu-id="9417a-124">If the <paramref name="innerException" /> parameter is not <see langword="null" />, the current exception is raised in a <see langword="catch" /> block that handles the inner exception.</span></span></param>
        <summary><span data-ttu-id="9417a-125">使用指定的错误消息和对作为此异常原因的内部异常的引用来初始化 <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> 类的新实例。</span><span class="sxs-lookup"><span data-stu-id="9417a-125">Initializes a new instance of the <see cref="T:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException" /> class with a specified error message and a reference to the inner exception that is the cause of this exception.</span></span></summary>
        <remarks>
          <format type="text/markdown"><![CDATA[  
  
## Remarks  
 <span data-ttu-id="9417a-126">作为前一个异常的直接结果引发的异常应在 <xref:System.Exception.InnerException%2A> 属性中包含对前一个异常的引用。</span><span class="sxs-lookup"><span data-stu-id="9417a-126">An exception that is thrown as a direct result of a previous exception should include a reference to the previous exception in the <xref:System.Exception.InnerException%2A> property.</span></span> <span data-ttu-id="9417a-127"><xref:System.Exception.InnerException%2A> 属性将返回传递给构造函数的同一值；如果 <xref:System.Exception.InnerException%2A> 属性不向构造函数提供内部异常值，则返回 `null`。</span><span class="sxs-lookup"><span data-stu-id="9417a-127">The <xref:System.Exception.InnerException%2A> property returns the same value that is passed into the constructor, or `null` if the <xref:System.Exception.InnerException%2A> property does not supply the inner exception value to the constructor.</span></span>  
  
 <span data-ttu-id="9417a-128">下表显示了 <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException> 实例的初始属性值。</span><span class="sxs-lookup"><span data-stu-id="9417a-128">The following table shows the initial property values for an instance of <xref:System.Data.Linq.ForeignKeyReferenceAlreadyHasValueException>.</span></span>  
  
|<span data-ttu-id="9417a-129">属性</span><span class="sxs-lookup"><span data-stu-id="9417a-129">Property</span></span>|<span data-ttu-id="9417a-130">值</span><span class="sxs-lookup"><span data-stu-id="9417a-130">Value</span></span>|  
|--------------|-----------|  
|<xref:System.Exception.InnerException%2A>|<span data-ttu-id="9417a-131">`null` 列中的一个值匹配。</span><span class="sxs-lookup"><span data-stu-id="9417a-131">`null`.</span></span>|  
|<xref:System.Exception.Message%2A>|<span data-ttu-id="9417a-132">`message` 中指定的错误消息字符串。</span><span class="sxs-lookup"><span data-stu-id="9417a-132">The error message string specified in `message`.</span></span>|  
  
 ]]></format>
        </remarks>
        <related type="Article" href="/dotnet/standard/exceptions/"><span data-ttu-id="9417a-133">处理和引发异常</span><span class="sxs-lookup"><span data-stu-id="9417a-133">Handling and Throwing Exceptions</span></span></related>
      </Docs>
    </Member>
  </Members>
</Type>
